scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -441,150 +428,106 @@ public long getRunningJobsCount() {
+            Collection<TaskStatus> taskStatus = Arrays.asList(TaskStatus.SUBMITTED,
+                                                              TaskStatus.PAUSED,
+                                                              TaskStatus.PENDING,
+                                                              TaskStatus.WAITING_ON_ERROR,
+                                                              TaskStatus.WAITING_ON_FAILURE);
+            Query query = session.getNamedQuery("getPendingTasksCount")
+                                 .setParameterList("jobStatus", NOT_FINISHED_JOB_STATUSES)
+                                 .setParameterList("taskStatus", taskStatus);
+
+            return (Long) query.uniqueResult();
         });
     }
 
     public long getRunningTasksCount() {
-        return executeReadOnlyTransaction(new SessionWork<Long>() {
-
-            @Override
-            public Long doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getRunningTasksCount")
-                                     .setParameterList("jobStatus", NOT_FINISHED_JOB_STATUSES)
-                                     .setParameterList("taskStatus", Arrays.asList(TaskStatus.RUNNING));
-
-                return (Long) query.uniqueResult();
-            }
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getRunningTasksCount")
+                                 .setParameterList("jobStatus", NOT_FINISHED_JOB_STATUSES)
+                                 .setParameterList("taskStatus", Arrays.asList(TaskStatus.RUNNING));
 
+            return (Long) query.uniqueResult();
         });
     }
 
     public long getTotalTasksCount() {
-        return executeReadOnlyTransaction(new SessionWork<Long>() {
-
-            @Override
-            public Long doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getTotalTasksCount");
-                Long count = (Long) query.uniqueResult();
-                return count;
-            }
-
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getTotalTasksCount");
+            Long count = (Long) query.uniqueResult();
+            return count;
         });
     }
 
     public double getMeanJobPendingTime() {
-        return executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query query = session.getNamedQuery("getMeanJobPendingTime");
-                Double result = (Double) query.uniqueResult();
-                return result == null ? 0 : result;
-            }
-
+        return executeReadOnlyTransaction(session -> {
+            Query query = session.getNamedQuery("getMeanJobPendingTime");
+            Double result = (Double) query.uniqueResult();
+            return result == null ? 0 : result;
```

Comment:
Why do you need to check for null here but not at the other queries above? 

I think it is better to change `0 : result` into `0d : result` for readability

Reply:
I did not want to touch any logic in this PR. I wanted just to transform some code in Java 8.